### PR TITLE
fix!(node/sled_store): Use `transaction` when more than one value get read

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: node/src/store.rs [hunk 1]
```diff
@@ -416,4 +426,523 @@ mod tests {
             assert!(to_headers_range(bound, last_index).unwrap().is_empty());
         }
     }
+
+    #[rstest]
+    #[case::in_memory(new_in_memory_store())]
```
[the remark this answers is not in the document]
Comment (zvolin) [reply node/src/store.rs:431]: but ofc not in scope of this change, if you don't want

**Review by zvolin — APPROVED** [review]
cool, one thing that I see is that maybe we could make those functions less verbose by dropping the transactional and by_db_key from their names, but I'm fine with not doing that too

File: node/src/store.rs [hunk 2]
```diff
@@ -129,8 +129,12 @@ pub trait Store: Send + Sync + Debug {
...
     ) -> Result<u64>;
 
-    /// Gets the sampling data for the height. `Ok(None)` indicates that sampling data
-    /// wasn't set in the store
+    /// Gets the sampling metadata for the height.
+    ///
+    /// `Err(StoreError::NotFound)` indicates that header **and** sampling metadata of
+    /// height are not in the store.
+    ///
+    /// `Ok(None)` indicates that header is in the store but sampling metadata are not set yet.
```
Comment (fl0rek) [inline node/src/store.rs:137]:
```suggestion
    /// `Err(StoreError::NotFound)` indicates that both header **and** sampling metadata for the requested
    /// height are not in the store.
    ///
    /// `Ok(None)` indicates that header is in the store but sampling metadata is not set yet.
```
wording niit

**Review by fl0rek — APPROVED** [review]
nice work, I'll be much easier to keep store behaviours consistent now 🎸

File: node/src/store.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (oblique) [reply node/src/store.rs:431]: yes it is appended on the name of the test

**Pull request merged**